fdbserver/fdbserver.actor.cpp

```diff
@@ -887,94 +887,91 @@ void restoreRoleFilesHelper(std::string dirSrc, std::string dirToMove, std::stri
 	}
 }
 
-int main(int argc, char* argv[]) {
-	try {
-		platformInit();
-		initSignalSafeUnwind();
-
-#ifdef ALLOC_INSTRUMENTATION
-		g_extra_memory = new uint8_t[1000000];
-#endif
-		registerCrashHandler();
-
-		// Set default of line buffering standard out and error
-		setvbuf(stdout, NULL, _IOLBF, BUFSIZ);
-		setvbuf(stderr, NULL, _IOLBF, BUFSIZ);
+namespace {
+enum Role {
+	Simulation,
+	FDBD,
+	Test,
+	MultiTester,
+	SkipListTest,
+	SearchMutations,
+	DSLTest,
+	VersionedMapTest,
+	CreateTemplateDatabase,
+	NetworkTestClient,
+	NetworkTestServer,
+	Restore,
+	KVFileIntegrityCheck,
+	KVFileGenerateIOLogChecksums,
+	ConsistencyCheck
+};
+struct CLIOptions {
+	std::string commandLine;
+	std::string fileSystemPath = "", dataFolder, connFile = "", seedConnFile = "", seedConnString = "", logFolder = ".",
+	            metricsConnFile = "", metricsPrefix = "";
+	std::string logGroup = "default";
+	Role role = FDBD;
+	uint32_t randomSeed = platform::getRandomSeed();
+
+	const char* testFile = "tests/default.txt";
+	std::string kvFile;
+	std::string testServersStr;
+	std::string whitelistBinPaths;
+	std::vector<std::string> publicAddressStrs, listenAddressStrs;
```

Comment:
As a general comment, would you mind grouping some fields together if possible. For example this `(public|listen)AddressStrs` can be next to `(public|listen)Addresses` below. Same could be said about fields regarding log files.

Reply:
Done